cmake/compiler_settings.cmake

```diff
@@ -70,6 +70,18 @@ else ()
 endif ()
 
 if (CMAKE_CXX_COMPILER_ID MATCHES GNU OR CMAKE_CXX_COMPILER_ID MATCHES Clang)
+  # Enforce -O0 in debug mode.
+  string(TOUPPER ${CMAKE_BUILD_TYPE} CMAKE_BUILD_TYPE_UPPER)
+  if (CMAKE_BUILD_TYPE_UPPER STREQUAL "DEBUG")
+    if (CMAKE_CXX_COMPILER_ID MATCHES GNU)
+      set (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -Og")
```

Comment:
Isn't this already the default set by CMake? Perhaps `-g` is different from `-Og`? I'm looking at [GNU.cmake](https://github.com/Kitware/CMake/blob/07cfb18f9d29cfc0588ede928846a03ec5599c48/Modules/Compiler/GNU.cmake#L44).

Reply:
Anand's reference https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html shows the detail. In short, -g turns on the debug info, and -O0 turns off optimization, which means 0 pass of optimization.